digiroad2-viite/src/test/scala/fi/liikennevirasto/viite/ProjectServiceSpec.scala

```diff
@@ -1709,6 +1709,116 @@ class ProjectServiceSpec extends FunSuite with Matchers with BeforeAndAfter {
+      val roadAddressLength = roadway.endAddrMValue - roadway.startAddrMValue
+      val projectId = Sequences.nextProjectId
+      val oldLength = 100.1
+      val linearLocationId = 1234l
+      val projectLink = ProjectLink(Sequences.nextViitePrimaryKeySeqValue, roadway.roadNumber, roadway.roadPartNumber,
+        roadway.track, roadway.discontinuity, roadway.startAddrMValue, roadway.endAddrMValue,
+        roadway.startAddrMValue, roadway.endAddrMValue, Some(DateTime.now().plusMonths(1)), None, Some("test"),
+        linkId, 0.0, oldLength, SideCode.TowardsDigitizing,
+        (Some(ProjectLinkCalibrationPoint(linkId, 0, 0, CalibrationPointSource.RoadAddressSource)),
+          Some(ProjectLinkCalibrationPoint(linkId, oldLength, roadway.endAddrMValue, CalibrationPointSource.RoadAddressSource))),
+        FloatingReason.NoFloating, Seq(Point(0.0, 0.0, 0.0), Point(0.0, oldLength, 0.0)), projectId, LinkStatus.Terminated,
+        roadway.roadType, LinkGeomSource.NormalLinkInterface, oldLength, roadway.id, linearLocationId, roadway.ely, false, None,
+        DateTime.now().minusMonths(10).getMillis(), roadway.roadwayNumber, roadway.roadName, Some(roadAddressLength),
+        Some(0), Some(roadway.endAddrMValue), Some(roadway.track), Some(roadway.roadNumber), Some(roadway.roadPartNumber)
+      )
+
+      // Create expiring road addresses
+      val roadAddressToHistory = Map(roadway.id -> RoadAddress(roadway.id, linearLocationId, roadway.roadNumber,
+        roadway.roadPartNumber, roadway.roadType, roadway.track, roadway.discontinuity, roadway.startAddrMValue,
+        roadway.endAddrMValue, Some(roadway.startDate), None, Some(roadway.createdBy), linkId, 0, oldLength,
+        SideCode.TowardsDigitizing, DateTime.now().minusMonths(20).getMillis(),
+        (Some(CalibrationPoint(linkId, 0, 0)), Some(CalibrationPoint(linkId, oldLength, oldEndAddr))),
+        FloatingReason.NoFloating, Seq(Point(0.0, 0.0, 0.0), Point(0.0, oldLength, 0.0)), LinkGeomSource.NormalLinkInterface,
+        roadway.ely, roadway.terminated, roadway.roadwayNumber, Some(roadway.validFrom), None, roadway.roadName
+      ))
+
+      // Check before change
+      roadwayDAO.fetchAllByRoadwayId(Seq(roadway.id)).head.validTo should be(None)
+
+      // Call the method to be tested
+      projectService.createHistoryRows(roadway.id, roadway, projectLink.startDate.get)
+
+      // Check results
+      roadwayDAO.fetchAllByRoadwayId(Seq(roadway.id)).head.validTo should be(None)
+      roadwayDAO.fetchAllByRoadwayId(Seq(roadway.id)).head.endDate should be(None)
+
+      // TODO In case of termination old roadway row must be expired and new one must be created with end date and termination code 1.
+      // Also previous history rows must get termination code 2 (old ones expired and new history rows must be created).
+      // Should we do it in createHistoryRows -method or somewhere else?
+
+    }
+  }
+
   //TODO this will be implemented at VIITE-1541
```

Comment:
Is this actually implemented in VIITE-1543?

Reply:
Yes, those tests will be implemented at VIITE-1543